src/inmanta/server/agentmanager.py

```diff
@@ -1295,6 +1267,75 @@ async def _fork_inmanta(
             if errhandle is not None:
                 errhandle.close()
 
+    async def _wait_for_agents(
```

Comment:
Moved this from an embedded function in `_ensure_agents` to a dedicated method. I made very limited changes: update the signature and the docstring, and I'll mark all other changes with a comment.